server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES93BloomFilterStoredFieldsFormat.java

```diff
@@ -270,21 +276,20 @@ public void writeField(FieldInfo info, BytesRef value) throws IOException {
         }
 
         private boolean isBloomFilterField(FieldInfo info) {
-            return (bloomFilterFieldInfo != null && bloomFilterFieldInfo.getFieldNumber() == info.getFieldNumber())
+            return (bloomFilterWriter != null && bloomFilterWriter.fieldInfo.getFieldNumber() == info.getFieldNumber())
                 || info.getName().equals(bloomFilterFieldName);
         }
 
-        private void addToBloomFilter(FieldInfo info, BytesRef value) {
+        private int getBloomFilterSizeInBits() {
+            var bloomFilterSizeInBits = defaultBloomFilterSizeInBitsSupplier.getAsInt();
+            assert isPowerOfTwo(bloomFilterSizeInBits) : "Bloom filter size is not a power of 2: " + bloomFilterSizeInBits;
+            return bloomFilterSizeInBits;
+        }
+
+        private void addToBloomFilter(FieldInfo info, BytesRef value) throws IOException {
             assert info.getName().equals(bloomFilterFieldName) : "Expected " + bloomFilterFieldName + " but got " + info;
-            bloomFilterFieldInfo = info;
-            var termHashes = hashTerm(value, hashes);
-            for (int hash : termHashes) {
-                final int posInBitArray = hash & (bloomFilterSizeInBits - 1);
-                final int pos = posInBitArray >> 3; // div 8
-                final int mask = 1 << (posInBitArray & 7); // mod 8
-                final byte val = (byte) (buffer.get(pos) | mask);
-                buffer.set(pos, val);
-            }
+            maybeInitializeBloomFilterWriter(info, getBloomFilterSizeInBits());
+            bloomFilterWriter.addToBloomFilter(info, value);
```

Comment:
nit: I would rename to:
```suggestion
            bloomFilterWriter.add(value);
```

Reply:
Done in https://github.com/elastic/elasticsearch/pull/137622/commits/07ad165b4088ad478873867476c8fdc00cf2777a